src/ui/components/Library/useFilters.ts

```diff
@@ -25,6 +25,40 @@ export const LibraryContext = createContext<LibraryContextType>({
   preview: null,
 });
 
+export function encodeFilter(str: string) {
+  let encodedStr = "";
+  while (str.length > 0) {
+    const [match] = str.match(/^(.*?:)/) || [];
+    if (match) {
+      encodedStr += match;
+      str = str.substring(match.length);
+
+      const nextChar = str.substring(0, 1);
+      if (["'", '"', "["].includes(nextChar)) {
+        const closingChar = nextChar == "[" ? "]" : nextChar;
```

Comment:
This is cute, but I think personally I would rather have an explicit object like

```
type Surrounder = {
  open: string;
  close: string;
};
const SURROUNDERS: Surrounder[] = [
  { open: "'", close: "'" },
  { open: '"', close: '"' },
  { open: "[", close: "]" },
];
```